src/mode/mode_handler.ts

```diff
@@ -6,16 +6,25 @@ import {Mode, ModeName} from './mode';
 import CommandMode from './mode_command';
 import InsertMode from './mode_insert';
 import VisualMode from './mode_visual';
+import {Mappings} from '../mapping/main';
+import * as util from '../util';
+import {CommandLineMode} from '../cmd_line/main';
 
 export default class ModeHandler {
     private modes : Mode[];
     private statusBarItem : vscode.StatusBarItem;
+    mappings : Mappings;
 
     constructor() {
+        this.mappings = new Mappings();
+                
         this.modes = [
             new CommandMode(),
             new InsertMode(),
             new VisualMode(),
+            new CommandLineMode(this.mappings),
+            // new OperatorPendingMode();
```

Comment:
I was putting some logic together for Operator commands, but have not thought about in this "mode" way. Was trying to do everything in command mode, this might be a better idea.